### PR TITLE
[ci] Add code security tasks to Azure DevOps pipeline

### PR DESCRIPTION
**Pull request by shiyu1994**
Add code security tasks to Azure DevOps pipeline, including CredScan, CodeQL (Semmle) and Component Governance Detection.

### CODE REVIEW

File: .vsts-ci.yml [hunk 1]
```diff
@@ -267,6 +269,7 @@ jobs:
         TASK: swig
       cpp_tests:
         TASK: cpp-tests
+  timeoutInMinutes: 480
```
Comment (StrikerRUS) [inline .vsts-ci.yml:272]: We are using Microsoft-hosted agent for Windows jobs, so this value cannot be bigger than `360` in our case.
> For 360 minutes (6 hours) on Microsoft-hosted agents with a public project and public repository
    https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#timeouts

**Comment by StrikerRUS** [comment]
I suggest splitting this PR into several ones according to the number of introduced code security tasks. It will greatly improve the reviewability of PR.

**Comment by shiyu1994** [comment]
Closed per the suggestion of @StrikerRUS. We'll decompose it into several part.

**Comment by shiyu1994** [comment]
Reopen to get the analysis log.

**Pull request closed without merging**